Classes/Task/OptimizeIndexTaskAdditionalFieldProvider.php

```diff
@@ -0,0 +1,281 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  The GNU General Public License can be found at
+ *  http://www.gnu.org/copyleft/gpl.html.
+ *
+ *  This script is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use ApacheSolrForTypo3\Solr\Backend\CoreSelectorField;
+use ApacheSolrForTypo3\Solr\Backend\SiteSelectorField;
+use ApacheSolrForTypo3\Solr\Domain\Site\Site;
+use ApacheSolrForTypo3\Solr\Domain\Site\SiteRepository;
+use TYPO3\CMS\Core\Localization\LanguageService;
+use TYPO3\CMS\Core\Messaging\FlashMessage;
+use TYPO3\CMS\Core\Page\PageRenderer;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Scheduler\AbstractAdditionalFieldProvider;
+use TYPO3\CMS\Scheduler\Controller\SchedulerModuleController;
+use TYPO3\CMS\Scheduler\Task\AbstractTask;
+use TYPO3\CMS\Scheduler\Task\Enumeration\Action;
+
+/**
+ * Adds an additional field to specify the Solr server to initialize the index queue for
+ *
+ * @author Jens Jacobsen <typo3@jens-jacobsen.de>
+ */
+class OptimizeIndexTaskAdditionalFieldProvider extends AbstractAdditionalFieldProvider
+{
+    /**
+     * Default language file of the extension linkvalidator
+     *
+     * @var string
+     */
+    protected $languageFile = 'LLL:EXT:solr/Resources/Private/Language/locallang.xlf';
+
+    /**
+     * Task information
+     *
+     * @var array
+     */
+    protected $taskInformation;
+
+    /**
+     * Scheduler task
+     *
+     * @var AbstractTask|ReIndexTask|NULL
+     */
+    protected $task = null;
+
+    /**
+     * Scheduler Module
+     *
+     * @var SchedulerModuleController
+     */
+    protected $schedulerModule;
+
+    /**
+     * Selected site
+     *
+     * @var Site
+     */
+    protected $site = null;
+
+    /**
+     * SiteRepository
+     *
+     * @var SiteRepository
+     */
+    protected $siteRepository;
+
+    /**
+     * @var PageRenderer
+     */
+    protected $pageRenderer = null;
+
+    /**
+     * ReIndexTaskAdditionalFieldProvider constructor.
+     */
+    public function __construct()
+    {
+        $this->siteRepository = GeneralUtility::makeInstance(SiteRepository::class);
+    }
+
+    /**
+     * @param array $taskInfo
+     * @param \TYPO3\CMS\Scheduler\Task\AbstractTask|NULL $task
+     * @param \TYPO3\CMS\Scheduler\Controller\SchedulerModuleController $schedulerModule
+     */
+    protected function initialize(
+        array $taskInfo = [],
+        AbstractTask $task = null,
+        SchedulerModuleController $schedulerModule
```

Comment:
The first two argument are optional, the third one is not.

```suggestion
    protected function initialize(
        SchedulerModuleController $schedulerModule,
        AbstractTask $task = null,
        array $taskInfo = []
```
Please change the order of arguments to solve this issue.

Reply:
I reordered the item.